hdl/QuestaSimScript.tmpl

```diff
@@ -1,86 +1,84 @@
 #!/bin/bash
 
-set -eu -o pipefail
-
-{{ $name := .Name }}
 {{ $outDir := .OutDir }}
 {{ $outScript := .OutScript }}
-{{ $partName := .PartName }}
-{{ $boardName := .BoardName }}
-{{ $verbose := .Verbose }}
-{{ $incDir := .IncDir }}
 
-rm -rf {{ .OutDir }}
-mkdir -p {{ .OutDir }}
-TMPDIR=$(mktemp -d -t ci-XXXXXXXXXX)
-
-(
-    cd ${TMPDIR}
-
-    {{ range .Ips }}
-    NAME=`basename {{ . }} .xci`
-    mkdir $NAME
-    cd $NAME
-    cp {{ . }} .
-
-    cat > generate.tcl << EOF
-set_part "{{ $partName }}"
-set_property "board_part"         "{{ $boardName}}" [current_project]
-set_property "target_language"    "Verilog"         [current_project]
-
-set ip $NAME.xci
-read_ip \$ip
-generate_target simulation [get_files \$ip]
-EOF
+set -eu -o pipefail
 
-    {{ if $verbose }}
-    vivado -mode batch -nolog -nojournal  -notrace -source generate.tcl
-    find -type f
-    {{ else }}
-    vivado -mode batch -nolog -nojournal  -notrace -source generate.tcl | ( grep -E "^(ERROR|WARNING|CRITICAL)" || true )
-    {{ end }}
-    cd ..
-    {{ end }}
+{{ if .Verbose }}
+set -x
+{{ end }}
 
-    cp -af . {{ .OutDir }}
-)
+rm -rf {{ .OutDir }}
+mkdir -p {{ .OutDir }}
 
 cat > {{ .OutScript }} <<EOF
 #!/bin/bash
 set -eu -o pipefail
 
-vlib work
-{{ range .Srcs }}vlog -svinputport=net +acc=rn -sv -nologo -quiet +incdir+{{ $incDir }} {{ . }}
+NEEDS_XSIM_GLBL=false
+EXTRA_MOD=
+SIMULATOR=questa
+SIM_FLAGS="-modelsimini {{ .OutDir }}/../modelsim.ini"
+SIM_SV_FLAGS="\${SIM_FLAGS} -svinputport=net +acc=rn -sv -nologo -quiet +incdir+{{ .IncDir }}"
+SIM_SV_COMPILER=vlog
+SIM_VHDL_FLAGS="\${SIM_FLAGS} -93"
+SIM_VHDL_COMPILER=vcom
+SIM_IP_DIR={{ .OutDir }}
+PROJECT_NAME={{ .Name }}
+
+{{ if ne .LibDir "" }}
+cp {{ .LibDir }}/modelsim.ini {{ .OutDir }}/..
+{{ else }}
+touch {{ .OutDir }}/modelsim.ini
```

Comment:
`{{ .OutDir }}/../modelsim.ini` ?

Why do you copy this?

Reply:
Because it's modified by the `vmap` command and I can only have one.

Reply:
I see. (note the missing `..` in the line starting with touch.)